wade-review.md

```diff
@@ -0,0 +1,596 @@
+## Getting started with Kafka Streams: The Processing Process
+
+
+Now, I'll pivot to building a Kafka Streams application that processes the 'github-pull_requests' events, creating an up-to-date ratio of open/closed pull requests.
+
+If you want to follow along with this part, check out the GitHub repo in the links below. I’m going to focus on the domain logic given our timeframe, so if you really want to dive in, I suggest consulting the codebase. 
+
+
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> {
+}
+```
+
+Now I'll  write the ProcessorSupplier, using the class provided by KafkaStreams. I'll use the record shape supplied by GitHubPRInfo as well as the object provided by GitHubPrStateCounter in our models folder.
+
+---
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+}
+```
+
+Our store key will be named 'state_counter'. 
+
+---
+
+```java
+static class MyProcessorSupplier implements ProcessorSupplier<String, GitHubPRInfo, String, GitHubPRStateCounter> { 
+	static final String STORE_KEY = "state-counter";
+   @Override
+   public Processor<String, GitHubPRInfo, String, GitHubPRStateCounter> get() {
+		return new Processor<>() {
```

Comment:
Is it worth discussing this in more detail? The first thing going through my head is that it seems weird to create a new instance on every call to "get". I'd be tempted to implement this more like a singleton. But maybe that's bad. If so, why?

Reply:
To answer I'll give some background context.  For each input topic partition, Kafka Streams assigns a task (the unit of work in Kafka Streams). Then you can have a `StreamThread` up to the number of tasks (you can have more, but they would be idle).  Each task gets a copy of the topology, so if you have a thread-per-task, it's critical that each one has its own processor instances.  FYI Kafka Streams also creates a separate state store per partition/task

Reply:
That makes sense. I assume that is at least in part to make sure things are thread safe.

It's unfortunate that it is done inside of a `get` method rather than a `create` method or similar. I wonder how often people do this incorrectly and end up with a mess.

Reply:
@Lucia I feel like this is an interesting point of discussion for the video.